### PR TITLE
[XM] - XM Mobile should not resolve from the GAC either

### PR DESCRIPTION
**Pull request by chamons**
- Centralize fix I made for XM 4.5 to apply to both and write larger test
- https://bugzilla.xamarin.com/show_bug.cgi?id=42001

### CODE REVIEW

File: msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.XM45.targets [hunk 1]
```diff
@@ -18,10 +18,6 @@ Copyright (C) 2015 Xamarin Inc. All rights reserved.
 	<!-- These hacks will need to be rewritten if/when we move from xbuild to msbuild. Most likely to use FrameworkPathOverride, which should hopefully work.-->
 	<!-- This lets us bypass the standard MSBuild framework resolution and point to our copy. -->
 	<!-- Without this, we'll resolve System.dll and such from the system mono  -->
-	<PropertyGroup Condition="!$(MonoBundlingExtraArgs.Contains('--allow-unsafe-gac-resolution'))" >
```
Comment (jstedfast) [inline msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.XM45.targets]: Do we still want to keep this XML comment? Or can it be removed now?

Comment (chamons) [inline msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.XM45.targets]: The other hacks in that file still apply 😢

Comment (jstedfast) [inline msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.XM45.targets]: ok

**Comment by jstedfast** [comment]
Looks good to me.

**Comment by xamarin-release-manager** [comment]
Build success

**Pull request merged**